lib/daru/dataframe.rb

```diff
@@ -173,7 +173,8 @@ def rows source, opts={}
         elsif ArrayHelper.array_of?(source, Vector)
           from_vector_rows(source, opts)
         else
-          raise ArgumentError, "Can't create DataFrame from #{source}"
+          row = source.map { |v| Array(v) }
```

Comment:
What would be consequences? What if `source` doesn't respond to `map`? 

And what if it _does_ respond, where are the tests for the change? Now not only `group_by` behavior affected.

Reply:
Have added a check for that case.

In terms of specs - I've added one for the breaking case and all others are still passing, do you have a particular spec case in mind?

Reply:
Not that. Rather, I believe that local bug should have local fix, e.g. either in `GroupBy#resultant_context` (where DF is tried to be created from "wrong" data), or in `DataFrame#access_row_tuples_by_indexs` (which probably should always return array-of-arrays, shouldn't it?)

Reply:
Ah I get it, agreed that's a better way. Have updated.